README.md

```diff
@@ -49,21 +51,49 @@ Please, read the [getting started](https://doc.bblf.sh/using-babelfish/getting-s
 import bblfsh
 
 client = bblfsh.BblfshClient("0.0.0.0:9432")
-uast = client.parse("/path/to/file.py").uast
-print(uast)
-# "filter' allows you to use XPath queries to filter on result nodes:
-print(bblfsh.filter(uast, "//Import[@roleImport and @roleDeclaration]//alias"))
-
-# filter\_[bool|string|number] must be used when using XPath functions returning
-# these types:
-print(bblfsh.filter_bool(uast, "boolean(//*[@strtOffset or @endOffset])"))
-print(bblfsh.filter_string(uast, "name(//*[1])"))
-print(bblfsh.filter_number(uast, "count(//*)"))
+ctx = client.parse("/path/to/file.py")
+print(ctx)
+# or to get the results in a dictionary:
+resdict = ctx.get_all()
 
-# You can also iterate on several tree iteration orders:
-it = bblfsh.iterator(uast, bblfsh.TreeOrder.PRE_ORDER)
+# "filter' allows you to use XPath queries to filter on result nodes:
+it = ctx.filter("//python:Call")
 for node in it:
-    print(node.internal_type)
+    print(node)
+    # or:
+    doSomething(node.get())
+
+# filter must be used when using XPath functions returning these types:
+# XPath queries can return different types (dicts, int, float, bool or str), 
+# calling get() with an item will return the right type, but if you must ensure
+# that you are getting the expected type (to avoid errors in the queries) there
+# are alterative typed versions:
+x = next(ctx.filter("boolean(//*[@strtOffset or @endOffset])").get_bool()
+y = next(ctx.filter("name(//*[1])")).get_str()
+z = next(ctx.filter("count(//*)").get_int() # or get_float()
+
+# You can also iterate using iteration orders different than the 
+# default preorder using the `iterate` method on `parse` result or node objects:
+
+# Directly over parse results
+iter = client.parse("/path/to/file.py").iterate(bblfsh.TreeOrder.POST_ORDER)
+for i in iter: ...
+
+# Over filter results (which by default are already iterators with PRE_ORDER):
+ctx = client.parse("file.py")
+newiter = ctx.filter("//python:Call").iterate(bblfsh.TreeOrder.LEVEL_ORDER)
+for i in newiter: ...
+
+# Over individual node objects to change the iteration order of
+# a specific subtree:
+ctx = client.parse("file.py")
+first_node = next(ctx)
+newiter = first_node.iterate(bblfsh.TreeOrder.POSITION_ORDER)
+for i in newiter: ...
+
+# You can also get the non semantic UAST or native AST:
+ctx = client.parse("file.py", mode=bblfsh.ModeDict["NATIVE"])
```

Comment:
We should use an enum class or define separate constants instead of raw strings.

Reply:
It's mapping the `protocol_v2_module.Mode.DESCRIPTOR.values_by_name` field. Maybe we could generate variable names from these strings using `eval` or `exec` but that's almost worse.